File: wrappers/php/lib/Kendo/UI/PivotConfiguratorMessages.php
<?php

namespace Kendo\UI;

class PivotConfiguratorMessages extends \Kendo\SerializableObject {
//>> Properties

    /**
    * The text messages displayed in the measure fields sections.
    * @param string $value
    * @return \Kendo\UI\PivotConfiguratorMessages
    */
    public function measures($value) {
        return $this->setProperty('measures', $value);
    }

    /**
    * The text messages displayed in the column fields sections.
    * @param string $value
    * @return \Kendo\UI\PivotConfiguratorMessages
    */
    public function columns($value) {
        return $this->setProperty('columns', $value);
    }

    /**
    * The text messages displayed in the row fields sections.
    * @param string $value
    * @return \Kendo\UI\PivotConfiguratorMessages
    */
    public function rows($value) {
        return $this->setProperty('rows', $value);
    }

    /**
    * The text messages displayed for measure label.
    * @param string $value
    * @return \Kendo\UI\PivotConfiguratorMessages
    */
    public function measuresLabel($value) {
        return $this->setProperty('measuresLabel', $value);
    }

    /**
    * The text messages displayed for rows label.
    * @param string $value
    * @return \Kendo\UI\PivotConfiguratorMessages
    */
    public function rowsLabel($value) {
        return $this->setProperty('rowsLabel', $value);
    }

    /**
    * The text messages displayed for column label.
    * @param string $value
    * @return \Kendo\UI\PivotConfiguratorMessages
    */
    public function columnsLabel($value) {
        return $this->setProperty('columnsLabel', $value);
    }

    /**
    * The text messages displayed for fields label.
    * @param string $value
    * @return \Kendo\UI\PivotConfiguratorMessages
    */
    public function fieldsLabel($value) {
        return $this->setProperty('fieldsLabel', $value);
    }

    /**
    * The text messages displayed in the field menu.
    * @param \Kendo\UI\PivotConfiguratorMessagesFieldMenu|array $value
    * @return \Kendo\UI\PivotConfiguratorMessages
    */
    public function fieldMenu($value) {
        return $this->setProperty('fieldMenu', $value);
    }

//<< Properties
}

?>

File: wrappers/mvc/src/Kendo.Mvc/UI/PivotConfigurator/PivotConfiguratorMessages.cs
namespace Kendo.Mvc.UI
{
    using Kendo.Mvc.Resources;
    using System.Collections.Generic;

    public class PivotConfiguratorMessages : JsonObject
    {
        private const string DefaultMeasures = "Drop Data Fields Here";
        private const string DefaultColumns = "Drop Column Fields Here";
        private const string DefaultRows = "Drop Rows Fields Here";
        private const string DefaultMeasuresLabel = "Measures";
        private const string DefaultColumnsLabel = "Columns";
        private const string DefaultRowsLabel = "Rows";
        private const string DefaultFieldsLabel = "Fields";

        public PivotConfiguratorMessages() 
        {
            Measures = Messages.PivotConfigurator_Measures;
            Columns = Messages.PivotConfigurator_Columns;
            Rows = Messages.PivotConfigurator_Rows;
            MeasuresLabel = Messages.PivotConfigurator_MeasuresLabel;
            ColumnsLabel = Messages.PivotConfigurator_ColumnsLabel;
            RowsLabel = Messages.PivotConfigurator_RowsLabel;
            FieldsLabel = Messages.PivotConfigurator_FieldsLabel;
            FieldMenu = new PivotFieldMenuMessages();
        }

        public string Measures { get; set; }
        public string Columns { get; set; }
        public string Rows { get; set; }
        public string MeasuresLabel { get; set; }
        public string ColumnsLabel { get; set; }
        public string RowsLabel { get; set; }
        public string FieldsLabel { get; set; }
        public PivotFieldMenuMessages FieldMenu { get; set; }

        protected override void Serialize(IDictionary<string, object> json)
        {
            if (Measures != DefaultMeasures)
            {
                json["measures"] = Measures;
            }

            if (Columns != DefaultColumns)
            {
                json["columns"] = Columns;
            }

            if (Rows != DefaultRows)
            {
                json["rows"] = Rows;
            }

            if (MeasuresLabel != DefaultMeasuresLabel)
            {
                json["measuresLabel"] = MeasuresLabel;
            }

            if (ColumnsLabel != DefaultColumnsLabel)
            {
                json["columnsLabel"] = ColumnsLabel;
            }

            if (RowsLabel != DefaultRowsLabel)
            {
                json["rowsLabel"] = RowsLabel;
            }

            if (FieldsLabel != DefaultFieldsLabel)
            {
                json["fieldsLabel"] = FieldsLabel;
            }

            var fieldMenu = FieldMenu.ToJson();
            if (fieldMenu.Count > 0)
            {
                json["fieldMenu"] = fieldMenu;
            }
        }
    }
}
